adapter/config/crd/conversion.go

```diff
@@ -82,17 +83,25 @@ func KabobCaseToCamelCase(s string) string {
 	for _, word := range words {
 		out = out + strings.Title(word)
 	}
+	r, _ := regexp.Compile(`^([^-]*?)rule(.*)$`)
```

Comment:
This is a dirty regex hack (eugh, sorry). Test cases updated to use real model types, but I doubt the logic here is sufficiently future-proof.

The original logic expects input in the format `route-rule-x` to output `RouteRuleX`. With the first hyphen removed in this PR (so `routerule-x` to `RouteRuleX`) there is no easy way to use the previous logic for general strings (as this now treats `somerule` and `somepolicy` specially).

Possible solutions: 
- enumerating the allowed types (probably best?) 
- using this regex (to be tidied up if so)
- another way?

Awaiting feedback then I'll refactor this into something more maintainable (is that's ever possible with a regex).

Reply:
@rshriram you sure you want to keep this regex hack?